File: Tarea6 (Ordenamiento por Inserción)/Tarea6.js
function insertionSort(arr) {
    for (let i = 1; i < arr.length; i++) {
        let temp = arr[i];
        let j = i - 1;
        while (j >= 0 && arr[j] > temp) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = temp;
    }
}

function printArr(arr) {
    console.log(arr.join(" "));
}

let arr = [70, 15, 2, 51, 60];

console.log("Antes de ordenar: ");
printArr(arr);

insertionSort(arr);

console.log("Después de ordenar: ");
printArr(arr);

File: Tarea6 (Ordenamiento por Inserción)/Tarea6.java
public class Tarea6 {
    public static void insertionSort(int[] arr) {
        for (int i = 1; i < arr.length; i++) {
            int temp = arr[i];
            int j = i - 1;
            while (j >= 0 && arr[j] > temp) {
                arr[j + 1] = arr[j];
                j--;
            }
            arr[j + 1] = temp;
        }
    }

    public static void printArr(int[] arr) {
        for (int val : arr) {
            System.out.print(val + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        int[] arr = {70, 15, 2, 51, 60};

        System.out.print("Antes de ordenar: ");
        printArr(arr);

        insertionSort(arr);

        System.out.print("Después de ordenar: ");
        printArr(arr);
    }
}
